src/main/java/com/google/devtools/build/lib/actions/Actions.java

```diff
@@ -369,4 +370,20 @@ public static GeneratingActions fromSingleAction(ActionAnalysisMetadata action)
       return actions;
     }
   }
+
+  /**
+   * Utility method to prefetch action inputs that turns any {@link IOException} into an
+   * {@link ActionExecutionException}.
+   */
+  public static void prefetchInputs(Iterable<? extends ActionInput> inputs,
```

Comment:
The documentation of `ActionInputPrefetcher#prefetchFiles()` says that the call does not block, i.e. when it returns, the files to be prefetched are not guaranteed to be present locally. Isn't that wrong?

Reply:
Do you think this code changes this contract?

(note: I am going to send a separate change that does update the API contract)

Reply:
AFAIU the actions you change rely on the fetched files being present, which is not guaranteed by the contract, don't they?

Do you know what `SpawnRunner#prefetchInputs()` (the only other place where this method is used) expects? I would guess that waiting for the files to be actually there would be at least some performance hit, since then you can't start an action before all its inputs have hit the local disk (instead of starting it immediately then waiting for an input only if it is actually opened)

Reply:
> AFAIU the actions you change rely on the fetched files being present, which is not guaranteed by the contract, don't they?

I updated the contract. Let's continue this discussion on the internal review.

Reply:
Well, I'm conflicted: on one hand, I like our internal review tool *much* better, but I also like staying out in the open. (In this case, I think what you are doing is perfectly fine and no further changes are required, so I fortunately don't have to make a choice now :) )